mutiny-core/src/nodemanager.rs

```diff
@@ -671,47 +653,33 @@ impl<S: MutinyStorage> NodeManager<S> {
 
         let nodes = Arc::new(Mutex::new(nodes_map));
 
-        let seed = mnemonic.to_seed("");
-        let xprivkey = ExtendedPrivKey::new_master(network, &seed)?;
-        let auth = AuthManager::new(xprivkey)?;
-
         let lnurl_client = Arc::new(
             lnurl::Builder::default()
                 .build_async()
                 .expect("failed to make lnurl client"),
         );
 
-        let auth_client = if let Some(auth_url) = c.auth_url {
-            let a = Arc::new(MutinyAuthClient::new(
-                auth.clone(),
-                lnurl_client.clone(),
-                logger.clone(),
-                auth_url,
-            ));
-            Some(a)
-        } else {
-            None
-        };
-
-        let subscription_client = if let Some(auth_client) = auth_client {
+        let (subscription_client, auth) = if let Some(auth_client) = c.auth_client {
             if let Some(subscription_url) = c.subscription_url {
+                let auth = auth_client.auth.clone();
                 let s = Arc::new(MutinySubscriptionClient::new(
                     auth_client,
                     subscription_url,
                     logger.clone(),
                 ));
-                Some(s)
+                (Some(s), auth)
             } else {
-                None
+                (None, auth_client.auth.clone())
             }
         } else {
-            None
+            let auth_manager = AuthManager::new(c.xprivkey)?;
+            (None, auth_manager)
         };
```

Comment:
Is this just so that the auth client can be passed in from `mutiny-wasm`? Maybe if that's required right now, we can just enforce that both auth manager and auth client be loaded in from there. Just don't like taking the inner of another object to use for something else.

Reply:
Yeah we could, felt weird to have to two optional copies of it so I did it like this, could do it the other way too